contracts/ModifiedTokenDistro.sol

```diff
@@ -0,0 +1,346 @@
+        require(balances[msg.sender].allocatedTokens > amount, 'TokenDistro::allocate NOT_ENOUGH_TOKENS');
+        balances[msg.sender].allocatedTokens = balances[msg.sender].allocatedTokens - amount;
+
+        balances[recipient].allocatedTokens = balances[recipient].allocatedTokens + amount;
+
+        if (claim && claimableNow(recipient) > 0) {
+            _claim(recipient);
+        }
+
+        emit Allocate(msg.sender, recipient, amount);
+    }
+
+    function allocate(address recipient, uint256 amount, bool claim) external override onlyDistributor {
+        _allocate(recipient, amount, claim);
+    }
+
+    /**
+     * Function that allows to the distributor address to allocate some amounts of tokens to specific recipients
+     * @dev Needs to be initialized: Nobody has the DEFAULT_ADMIN_ROLE and all available tokens have been assigned
+     * @param recipients of token allocation
+     * @param amounts allocated amount
+     *
+     * Unlike allocate method it doesn't claim recipients available balance
+     */
+    function _allocateMany(address[] memory recipients, uint256[] memory amounts) internal onlyDistributor {
+        uint256 length = recipients.length;
+        require(length == amounts.length, 'TokenDistro::allocateMany: INPUT_LENGTH_NOT_MATCH');
+
+        for (uint256 i = 0; i < length; i++) {
+            _allocate(recipients[i], amounts[i], false);
+        }
+    }
+
+    function allocateMany(address[] memory recipients, uint256[] memory amounts) external override {
+        _allocateMany(recipients, amounts);
+    }
+
+    function sendGIVbacks(address[] memory recipients, uint256[] memory amounts) external override {
+        _allocateMany(recipients, amounts);
+        emit GivBackPaid(msg.sender);
+    }
+
+    function sendPraiseRewards(address[] memory recipients, uint256[] memory amounts) external {
+        _allocateMany(recipients, amounts);
+        emit PraiseRewardPaid(msg.sender);
+    }
+
+    /**
+     * Function that allows a recipient to change its address
+     * @dev The change can only be made to an address that has not previously received an allocation &
+     * the distributor cannot change its address
+     *
+     * Emits a {ChangeAddress} event.
+     *
+     */
+    function changeAddress(address newAddress) external override {
+        _transferAllocation(msg.sender, newAddress);
+    }
+
+    /**
+     * Function to get the current timestamp from the block
+     */
+    function getTimestamp() public view virtual override returns (uint256) {
+        return block.timestamp;
+    }
+
+    /**
+     * Function to get the total claimable tokens at some moment
+     * @param timestamp Unix time to check the number of tokens claimable
+     * @return Number of tokens claimable at that timestamp
+     */
+    function globallyClaimableAt(uint256 timestamp) public view override returns (uint256) {
+        require(duration > 0, 'TokenDistro::globallyClaimableAt: DURATION_ZERO');
```

Comment:
No need to do this. In case the duration be zero, this division will cause tx revert
```
return initialAmount + (deltaTime * lockedAmount) / duration;
```